miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/database/MiniAppSecureDatabase.kt

```diff
@@ -4,12 +4,14 @@ import android.annotation.SuppressLint
 import android.content.ContentValues
 import android.content.Context
 import android.database.sqlite.SQLiteDatabase
-import android.database.sqlite.SQLiteException
 import androidx.annotation.NonNull
 import androidx.sqlite.db.SupportSQLiteDatabase
+import net.sqlcipher.database.SQLiteFullException
 import java.io.IOException
 import java.sql.SQLException
+import java.util.stream.Collectors
 
+private const val DB_BATCH_SIZE = 100
```

Comment:
Can we have something dynamic batch size here ? Like in [iOS](https://github.com/rakutentech/ios-miniapp/pull/421/files#diff-b9d5290749067641f8dcb160aa1d2734283aa46a9a5ceab4fc6b6b56e9a1d2f7R133)

Reply:
@rleojoseph Well, I looked in to the `iOS` code and I have a feeling that it's in fact not dynamic. It seems in iOS The batch size is being decided based on the `dict` count which means if the dict count is between `5000..<25000` then the whole data will be written in batches of `100` items it's because dict count will be static and the batch size won't be changed during the `insert`. It would have been dynamic if the data is being `inserted` with different batch sizes in runtime and if you notice the code closely then you would find that it's kinda not possible at least for Android.

Therefore if we follow the same logic in `Android` bases on dict count. It'll again raise that same size issue. The page size will increase and eventually it'll leave the more size for partial data insert which I had shown you earlier.

Reply:
<img width="781" alt="Screen Shot 2022-07-08 at 12 26 17" src="https://user-images.githubusercontent.com/79963210/177911205-7a3f66b3-d318-4e44-8e4d-be3ffd83d0d4.png">

Reply:
@gopi-karmakar , I agree. It's not dynamic but definitely better than having a fixed batch size. Having a fixed less batch size would definitely have performance issues when the number of records increases close to 100k. 
This is not a blocker though, but a suggestion. Because you will see a noticeable performance delay on this.